pyanaconda/modules/storage/partitioning/manual_interface.py

```diff
@@ -0,0 +1,92 @@
+# Red Hat, Inc.
+#
+from pyanaconda.core.constants import MOUNT_POINT_PATH, MOUNT_POINT_DEVICE, MOUNT_POINT_FORMAT, \
+    MOUNT_POINT_REFORMAT, MOUNT_POINT_FORMAT_OPTIONS, MOUNT_POINT_MOUNT_OPTIONS
+from pyanaconda.dbus.interface import dbus_interface
+from pyanaconda.dbus.property import emits_properties_changed
+from pyanaconda.dbus.typing import *  # pylint: disable=wildcard-import
+from pyanaconda.modules.common.base import KickstartModuleInterfaceTemplate
+from pyanaconda.modules.common.constants.objects import MANUAL_PARTITIONING
+
+
+@dbus_interface(MANUAL_PARTITIONING.interface_name)
+class ManualPartitioningInterface(KickstartModuleInterfaceTemplate):
+    """DBus interface for the manual partitioning module."""
+
+    def connect_signals(self):
+        """Connect the signals."""
+        super().connect_signals()
+        self.watch_property("Enabled", self.implementation.enabled_changed)
+        self.watch_property("MountPoints", self.implementation.mount_points_changed)
+
+    @property
+    def Enabled(self) -> Bool:
+        """Is the manual partitioning enabled?"""
+        return self.implementation.enabled
+
+    @emits_properties_changed
+    def SetEnabled(self, enabled: Bool):
+        """Is the manual partitioning enabled?
+
+        :param enabled: True if enabled, otherwise False
+        """
+        self.implementation.set_enabled(enabled)
+
+    @property
+    def MountPoints(self) -> List[Dict[Str, Variant]]:
+        """List of mount point assignments."""
+        return [{
+            MOUNT_POINT_PATH: get_variant(Str, point.mount_point),
+            MOUNT_POINT_DEVICE: get_variant(Str, point.device),
+            MOUNT_POINT_REFORMAT: get_variant(Bool, point.reformat),
+            MOUNT_POINT_FORMAT: get_variant(Str, point.new_format),
+            MOUNT_POINT_FORMAT_OPTIONS: get_variant(Str, point.format_options),
+            MOUNT_POINT_MOUNT_OPTIONS: get_variant(Str, point.mount_options)
+        } for point in self.implementation.mount_points]
+
+    @emits_properties_changed
+    def SetMountPoints(self, mount_points: List[Dict[Str, Variant]]):
+        """Set the mount point assignments."""
+        points = []
+
+        for data in mount_points:
+            point = self.implementation.get_new_mount_point()
+
+            if MOUNT_POINT_PATH in data:
+                point.set_mount_point(data[MOUNT_POINT_PATH])
+
+            if MOUNT_POINT_DEVICE in data:
+                point.set_device(data[MOUNT_POINT_DEVICE])
+
+            if MOUNT_POINT_REFORMAT in data:
+                point.set_reformat(data[MOUNT_POINT_REFORMAT])
+
+            if MOUNT_POINT_FORMAT in data:
+                point.set_new_format(data[MOUNT_POINT_FORMAT])
+
+            if MOUNT_POINT_FORMAT_OPTIONS in data:
+                point.set_format_options(data[MOUNT_POINT_FORMAT_OPTIONS])
+
+            if MOUNT_POINT_MOUNT_OPTIONS in data:
+                point.set_mount_options(data[MOUNT_POINT_MOUNT_OPTIONS])
```

Comment:
What about avoiding copy & paste by:
```
self._set_mountpoint_value(point.set_mount_point, MOUNT_POINT_PATH, data)
self._set_mountpoint_value(point.set_device, MOUNT_POINT_DEVICE, data)
....

def _set_mountpoint_value(prop_func, value, data):
    if value in data:
        prop_func(data[value])
```

Or even better to have fill_up method which will take DBus formatted data and create the object from it. It can be than shared with the UI code to make the life easier for us.

Reply:
This will be fixed once the support for DBus structures is implemented.